olp-cpp-sdk-core/include/olp/core/client/OlpClient.h

```diff
@@ -160,6 +160,33 @@ class CORE_API OlpClient {
                        RequestBodyType post_body, std::string content_type,
                        CancellationContext context) const;
 
+  /**
+   * @brief Executes the HTTP request through the network stack in a blocking
+   * way. The response content is consumed via data callback.
+   *
+   * @param path The path that is appended to the base URL.
+   * @param method Select one of the following methods: `GET`, `POST`, `DELETE`,
+   * or `PUT`.
+   * @param query_params The parameters that are appended to the URL path.
+   * @param header_params The headers used to customize the request.
+   * @param data_callback The network data callback to retrieve content.
+   * @param post_body For the `POST` request, populate `form_params` or
+   * `post_body`, but not both. This data must not be modified until
+   * the request is completed.
+   * @param content_type The content type for the `post_body` or `form_params`.
+   * @param context The `CancellationContext` instance that is used to cancel
+   * the request.
+   *
+   * @return The `HttpResponse` instance.
+   */
+  HttpResponse CallApiStream(std::string path, std::string method,
```

Comment:
Can we use const std::string& instead of string copying for path, method and content_type ?

Reply:
Same for non-integrated types if any. (ParametersType maybe)

Reply:
I would argue not to do that since the current approach is far more versatile than with const-ref because the current approach is friendly to in-place constructed objects since you could just move them, while const-ref should be copied later.